posthog.go

```diff
@@ -245,21 +259,28 @@ func (c *client) ReloadFeatureFlags() error {
 	return nil
 }
 
-func (c *client) GetFeatureFlag(flagKey string, distinctId string, defaultValue interface{}) (interface{}, error) {
+func (c *client) GetFeatureFlag(flagConfig FeatureFlagPayload) (interface{}, error) {
+	if err := flagConfig.validate(); err != nil {
```

Comment:
Major change is that we accept a featureflag payload and validate it on invocation. This way users don't need to fill in every argument everytime